client.go

```diff
@@ -571,10 +571,13 @@ func (client *client) update(data *MetadataResponse) ([]string, error) {
 				toRetry[topic.Name] = true
 			}
 		}
-		var partitionCache [maxPartitionIndex][]int32
-		partitionCache[allPartitions] = client.setPartitionCache(topic.Name, allPartitions)
-		partitionCache[writablePartitions] = client.setPartitionCache(topic.Name, writablePartitions)
-		client.cachedPartitionsResults[topic.Name] = partitionCache
+
+		if topic.Err != ErrUnknownTopicOrPartition && topic.Err != ErrInvalidTopic {
```

Comment:
Additional thought is that we still create the `client.metadata` map entry even if the cache is gone, which could be a little confusing (that the "real" data is `[]` but the cache is `nil`)